### PR TITLE
Add the kafka-metrics-exporter crate

### PR DESCRIPTION
**Pull request by colin-grapl**
### Which issue does this PR correspond to?
issue-tracker-514

### What changes does this PR make to Grapl? Why?
The kafka-metrics-exporter is a simple crate. It implements an interface defined in the `metrics` crate and adds logic to push those metrics out to Kafka. It supports counters, gauges, and histograms.

The metrics messages are themselves encoded as protobuf (see graplinc.grapl.api.metrics.metric_message).

Using the crate is fairly straightforward:

```rust
        let producer: FutureProducer = producer_init()?;
        KafkaMetricExporterBuilder::new(
            "metrics-topic",
            producer,
        ).install()?;

        histogram!("process.query_time", 1234f64);
        counter!("process.query_row_count", 1000);
```

The exporter will set up a background thread that pushes the metrics off to Kafka.

### How were these changes tested?
There's an integration test that spins up Kafka and emits metrics to it, and then consumes those metrics.

### CODE REVIEW

File: src/rust/kafka-metrics-exporter/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+            message="register_gauge - does nothing for KafkaMetricExporter",
+            name=&key.name(),
+            unit=?unit,
+            description=?description,
+        );
+    }
+
+    fn register_histogram(&self, key: &Key, unit: Option<Unit>, description: Option<&'static str>) {
+        tracing::trace!(
+            message="register_histogram - does nothing for KafkaMetricExporter",
```
Comment (christophermaier) [inline src/rust/kafka-metrics-exporter/src/lib.rs:208]: Why can we not register things, but can update them?

Comment (colin-grapl) [reply src/rust/kafka-metrics-exporter/src/lib.rs:208]: Registering is just where you set descriptions for metrics, for systems like prometheus I think this is more important. For this type of metric there's nothing we'd want to register - any registration is going to have to happen out of band.

File: src/proto/graplinc/grapl/metrics/v1/metric_types.proto [hunk 2]
```diff
@@ -0,0 +1,73 @@
+syntax = "proto3";
+
+package graplinc.grapl.metrics.v1;
```
[the remark this answers is not in the document]
Comment (colin-grapl) [reply src/proto/graplinc/grapl/metrics/v1/metric_types.proto:3]: Good question - I basically copied the schemas from the rust metrics crate 1:1, but I should look to see.

File: src/proto/graplinc/grapl/metrics/v1/metric_types.proto [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    string value = 2;
+}
+
+// Counters represent a single monotonic value, which means the value can only be incremented,
+// not decremented, and always starts out with an initial value of zero.
+message Counter {
+    // The name of the metric
+    string name = 1;
+    // The value to increment the counter by
+    uint64 value = 2;
```
[the remark this answers is not in the document]
Comment (colin-grapl) [reply src/proto/graplinc/grapl/metrics/v1/metric_types.proto:21]: I just copied the schema from the metrics crate - increment does make sense.

**Review by jgrillo-grapl — APPROVED** [review]
👍

**Pull request merged**